### ISSUE TITLE
Replace world_physical_axes_types with property giving which world axes are correlated with pixel axes.

### ISSUE THREAD

**Issue by DanRyanIrish** [sunpy/ndcube#220]
### Description

```NDCube.world_axis_physical_types``` only returns the world types of the WCS axes.  It has no knowledge of pixel axes and now also exists on the WCS object itself.  Suggest removing it from ```NDCube``` and access it through the WCS in future.

The original aim of ```world_axis_physical_types``` from an ```NDCube``` point of view was to convey which world axes corresponded to which pixel axes.  Add a new property that provides this info.  It could be expressed as a table or tuple of tuples?
An example is a cube with lon, lat, wavelength.  The output of this property could be ```((lon, lat), (lon, lat), wave))``` or a table giving the same info.

**Comment by DanRyanIrish** [comment]
@Cadair, this issue seems relevant to our discussion earlier today.

Also, if we adopt it, we should include it in 2.0 as it would be reinterpretation or outright break in the API.

**Comment by dstansby** [comment]
Saw this in passing, and thought I'd quickly say that [`BaseLowLevelWCS.axis_correlation_matrix` ](https://astropy.readthedocs.io/en/stable/api/astropy.wcs.wcsapi.BaseLowLevelWCS.html#astropy.wcs.wcsapi.BaseLowLevelWCS.axis_correlation_matrix) in astropy seems to be the easy way to achieve this.

**Comment by DanRyanIrish** [comment]
Thanks @dstansby.  That certainly seems the way to do it.  Having thought a little more, we could make this a new method, ```array_axis_physical_types``` and deprecate ```world_axis_physical_types``` in favour of ```wcs.world_axis_physical_types```.

Here's how I see this could be done.  ```array_axis_physical_types``` would give the physical types for each array axis while an inverse method could give the array axes for each physical type.  Say we call that ```physical_type_array_axes```.  Then they could be implemented like so:
```python
@property
def array_axis_physical_types(self):
    return tuple([tuple(self.wcs.world_axis_physical_types[self.wcs.axis_correlation_matrix[:, i]]) for i in range(self.wcs.pix_n_dim)])

@property
def physical_type_array_axes(self):
    pix_dims = np.arange(n_pix_dim)
    dict([(self.wcs.world_axis_physical_types[i], tuple(pix_dims[self.wcs.axis_correlation_matrix[i]])) for i in range(self.wcs.world_n_dim)])
```

These would then return outputs like the following:
```python
>>> self.array_axis_physical_types
(('custom:pos.helioprojective.lon', 'custom:pos.helioprojective.lat', 'time'),
 ('custom:pos.helioprojective.lon', 'custom:pos.helioprojective.lat'),
 ('em.wl',),
 ('time',))

>>> self.physical_type_array_axes
{'custom:pos.helioprojective.lon': (0, 1),
 'custom:pos.helioprojective.lat': (0, 1),
 'em.wl': (2,),
 'time': (0, 3)}
```

Two questions arise from this:  should, what is essentially a convenience function on the ```wcs.axis_correlation_matrix```, be provided?  And if so, should it be provided at the ```NDCube``` level, or on the ```wcs``` itself?  An argument for this ```NDCube```, when thinking of the current implementation, is that only at the ```NDCube``` level can ```missing_axes``` be accounted for.  We need this in particular for slicing.  However, if we will rely on ```SlicedWCS``` infrastructure for ```ndcube2.0```, this reasoning is no longer valid.  On reflection, if this functionality should exist, it should probably be in the ```wcs``` since ```wcs```'s can no be sliced and that process updates the correlation matrix accordingly.

Thoughts?  @Cadair?

**Comment by DanRyanIrish** [comment]
This issue was reraised by #240.  See there for an alternative but similar implementation.

**Comment by DanRyanIrish** [comment]
Addressed by #281
